lib/content-services/search/search-query-builder.service.ts

```diff
@@ -116,12 +116,31 @@ export class SearchQueryBuilderService {
                 fields: this.config.fields,
                 filterQueries: this.filterQueries,
                 facetQueries: this.config.facetQueries,
-                facetFields: this.config.facetFields
+                facetFields: this.facetFields
             };
 
             return result;
         }
 
         return null;
     }
+
+    private get facetFields(): RequestFacetFields {
+        const facetFields = this.config.facetFields;
+
+        if (facetFields && facetFields.length > 0) {
+            return {
+                facets: facetFields.map(facet => <RequestFacetField> {
+                    field: facet.field,
+                    mincount: facet.mincount,
+                    label: facet.label,
+                    limit: facet.limit,
+                    offset: facet.offset,
+                    prefix: facet.prefix
+                })
+            };
+        }
+
+        return null;
```

Comment:
is that needed return null? doesn't propagate null usually is a good thing

Reply:
this is needed to set `null` for the QueryBody property, otherwise it may have unexpected results.